### PR TITLE
Pt 2664 collaborators

### PR DESCRIPTION
**Pull request by a-gvo**
@itaiGershtansky @marta-  Please review.

### CODE REVIEW

**Comment by marta-** [comment]
@a-gvo first comment: Please use the issue ID  exactly as it appears in JIRA ("PT-2264" in this case, including the dash), otherwise the PR might not show up in the Development box on JIRA for the target issue.

File: components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml [hunk 1]
```diff
@@ -1104,6 +1104,20 @@ $xwiki.ssx.use($serviceDocName)
   #set ($visibility =  $patientAccess.visibility)
   #set ($collaborators = $patientAccess.collaborators)
   #set ($collaboratorsCount = $collaborators.size())
+  #set($collaboratorString = 'my collaborator string')
```
Comment (itaiGershtansky) [inline components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml]: That was just for debug, should be an empty string.

File: components/patient-access-rules/api/src/main/resources/ApplicationResources.properties [hunk 2]
```diff
@@ -33,7 +33,10 @@ phenotips.patientAccessRightsManagement.heading=Access rights for __patientID__
 phenotips.patientAccessRightsManagement.noOwner=nobody
 phenotips.patientAccessRightsManagement.selfOwner=you
 phenotips.patientAccessRightsManagement.caseOwnerandVisibility=This case is owned by {0} and is {1}.
-phenotips.patientAccessRightsManagement.collaboratorsSummary=It is shared with {0} {1,choice,0#no collaborators|1#1 collaborator|1<{1} collaborators} {2}
+phenotips.patientAccessRightsManagement.collaboratorsSummary=It is shared with
+phenotips.patientAccessRightsManagement.collaboratorsSummary.noCollaborators=no collaborators
+phenotips.patientAccessRightsManagement.collaboratorsSummary.oneCollaborator={0} and {1} other collaborator
+phenotips.patientAccessRightsManagement.collaboratorsSummary.xCollaborators={0} and {1} other collaborators
 phenotips.patientAccessRightsManagement.modifyButton=Modify permissions
```
Comment (marta-) [inline components/patient-access-rules/api/src/main/resources/ApplicationResources.properties]: There's no guarantee that this breakdown into pieces will work in some other languages. It certainly won't work in German, where the word for "shared" needs to go at the very end of the sentence (the order of the words would be "It is with {.....} shared"). So always translate full sentences, not pieces if sentences. If the variation with the parameters is too complex, use different keys and do `if`s in velocity.

File: components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml [hunk 3]
```diff
@@ -1137,7 +1151,7 @@ $xwiki.ssx.use($serviceDocName)
     $services.localization.render("phenotips.patientAccessRightsManagement.caseOwnerandVisibility",["(% class='access-rights-info' %)$ownerElement(%%)","(% class='access-rights-info visibility' title='$!{visibility.description}' %){{icon name='$visibility'/}}${visibility.label}(%%)"])##
     ##if ($canManage),#else and#end it is (% class="access-rights-info visibility" title="$!{visibility.description}" %){{icon name="$visibility"/}}${visibility.label}(%%)##
     #if ($canManage) ##
-      $services.localization.render("phenotips.patientAccessRightsManagement.collaboratorsSummary",["(% class='access-rights-info' %){{html wiki='true' clean='false'}}{{icon name='group'/}}&lt;span class='collaborators'&gt;",$collaboratorsCount, "&lt;/span&gt;{{/html}}(%%)"])##
+            $services.localization.render("phenotips.patientAccessRightsManagement.collaboratorsSummary")(% class='access-rights-info' %){{html wiki='true' clean='false'}}{{icon name='group'/}}&lt;span class='collaborators'&gt;$collaboratorString&lt;/span&gt;{{/html}}(%%)##
```
Comment (itaiGershtansky) [inline components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml]: Indentation

File: components/patient-access-rules/api/src/main/resources/ApplicationResources.properties [hunk 4]
```diff
@@ -33,7 +33,10 @@ phenotips.patientAccessRightsManagement.heading=Access rights for __patientID__
 phenotips.patientAccessRightsManagement.noOwner=nobody
 phenotips.patientAccessRightsManagement.selfOwner=you
 phenotips.patientAccessRightsManagement.caseOwnerandVisibility=This case is owned by {0} and is {1}.
-phenotips.patientAccessRightsManagement.collaboratorsSummary=It is shared with {0} {1,choice,0#no collaborators|1#1 collaborator|1<{1} collaborators} {2}
+phenotips.patientAccessRightsManagement.collaboratorsSummary=It is shared with
+phenotips.patientAccessRightsManagement.collaboratorsSummary.noCollaborators=no collaborators
+phenotips.patientAccessRightsManagement.collaboratorsSummary.oneCollaborator={0} and {1} other collaborator
+phenotips.patientAccessRightsManagement.collaboratorsSummary.xCollaborators={0} and {1} other collaborators
```
Comment (marta-) [inline components/patient-access-rules/api/src/main/resources/ApplicationResources.properties]: There's no guarantee that this breakdown into pieces will work in some other languages. It certainly won't work in German, where the word for "shared" needs to go at the very end of the sentence (the order of the words would be "It is with {.....} shared"). So always translate full sentences, not pieces if sentences. If the variation with the parameters is too complex, use different keys and do `if`s in velocity. This is a blocker issue.

File: components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml [hunk 5]
```diff
@@ -736,12 +736,16 @@
...
+                var topCollaborator = $$('#manage-collaborators tr').map(function(item) {return item.down('select').selectedIndex + ':' + item.down('td:nth-child(2)').innerText}).max();
+                topCollaborator = topCollaborator.substring(topCollaborator.indexOf(':') + 1);
+                var collaboratorText = topCollaborator;
+                if (collaboratorsCount &gt; 0) {
+                  collaboratorText = collaboratorsCount == 1 ? "$services.localization.render('phenotips.patientAccessRightsManagement.collaboratorsSummary.oneCollaborator', ['" + collaboratorText + "', '" + collaboratorsCount + "'])" : "$services.localization.render('phenotips.patientAccessRightsManagement.collaboratorsSummary.xCollaborators', ['" + collaboratorText + "', '" + collaboratorsCount + "'])";
```
Comment (itaiGershtansky) [inline components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml]: I think it's better to use ? for shorter expressions and put an explicit if here.

**Pull request closed without merging**